src/Components/EditPost.js

```diff
@@ -0,0 +1,79 @@
+import React, {useState} from "react";
+import {useParams} from "react-router";
+import {CREATE_POST_INPUT, UPDATE_POST} from "../GraphQL/Mutations";
+import {LOAD_POST} from "../GraphQL/Queries";
+import {useMutation, useQuery} from "@apollo/client";
+import { Field, Form, Formik } from 'formik';
+import {validateBody, validateTitle}  from "../shared/validation"
+
+function AddEditPost(){
+    const { id }= useParams();
+    const [success, setSuccess] = useState('')
+    const {loading, data} = useQuery(LOAD_POST, {
+        variables : {
+            id: id
+        },
+        skip: !id
+    })
+    const [updatePost] = useMutation(UPDATE_POST, {
+        onError: (error) => {
+            setSuccess(error.message)
+        },
+        onCompleted: () => {
+           setSuccess('Request was successfully completed!')
+    }
+    });
+    const [createPost ] = useMutation(CREATE_POST_INPUT, {
+        onError: (error) => {
+            setSuccess(error.message)
+        },
+        onCompleted: () => {
+            setSuccess('Request was successfully completed!')
+        }
+    })
+
+    function onSubmit(values){
+            if (id) {
+                updatePost({
+                        variables: {
+                            id: id,
+                            input: {
+                                ...values
+                            }
```

Comment:
It's better to use 2 spaces instead of 4

Reply:
fixed